### PR TITLE
TINY-4864: We now do not create dropdowns if there's just one entry f…

### PR DESCRIPTION
**Pull request by HAFRMO**
…or advlist split buttons.

### CODE REVIEW

**Comment by lnewson** [comment]
The JIRA number for this also appears to be incorrect, as this TINY-4864 is the training epic.

**Comment by lnewson** [comment]
Another issue I see is that this doesn't update the tooltip for the button based on the specified style. It also maybe should change the icon as well, though I'd check with the rest of the team and maybe design on that.

**Comment by metricjs** [comment]
FYI CLA is now complaining because I did a commit (Hannes and I paired to write the first half of a test) and the CLA system doesn't like me

File: modules/tinymce/src/plugins/advlist/test/ts/browser/ToolbarButtonStructureTest.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        menubar: false,
+        statusbar: false,
+        base_url: '/project/tinymce/js/tinymce'
+      }),
+      NamedChain.asChain([
+        NamedChain.direct(NamedChain.inputName(), Chain.identity, 'editor'),
+        NamedChain.direct('editor', cGetContainer, 'editorContainer'),
+        NamedChain.direct('editorContainer', Waiter.cTryUntil('', Assertions.cAssertStructure(
+          'Check p>strong element path',
+          ApproxStructure.build((s, str, arr) => s.element('div', {
```
Comment (lnewson) [inline modules/tinymce/src/plugins/advlist/test/ts/browser/ToolbarButtonStructureTest.ts:32]: This is way more than what's needed, as you don't need to assert the entire editor structure. So you could simplify all this `NamedChain` code to be at least half the size:

```
    Log.chainsAsStep('TBA', 'Test that one list type = toolbar button NOT splitbutton', [
      Editor.cFromSettings({
        theme: 'silver',
        plugins: 'lists advlist',
        toolbar: 'numlist',
        advlist_number_styles: 'default',
        menubar: false,
        statusbar: false,
        base_url: '/project/tinymce/js/tinymce'
      }),
      NamedChain.asChain([
        NamedChain.writeValue('body', Body.body()),
        NamedChain.direct('body', UiFinder.cFindIn('.tox-editor-header .tox-toolbar .tox-toolbar__group'), 'toolbarGroup'),
        NamedChain.read('toolbarGroup', Waiter.cTryUntil('', Assertions.cAssertStructure(
          'Check lists toolbar button structure',
          ApproxStructure.build((s, str, arr) => s.element('div', {
            classes: [ arr.has('tox-toolbar__group') ],
            children: [
              s.element('button', {
                classes: [ arr.has('tox-tbtn') ],
                attrs: {
                  title: str.is('Numbered list')
                }
              })
            ]
          }))
        ))),
        NamedChain.outputInput
      ]),
      Editor.cRemove
    ])
```

Comment (lnewson) [reply modules/tinymce/src/plugins/advlist/test/ts/browser/ToolbarButtonStructureTest.ts:32]: Also worth noting the editor wasn't being cleaned up in the previous code, so the above handles that.

**Comment by HAFRMO** [comment]
Checked the posibility of changing the icon. While quite doable the icons we could change to would not match the other icons due to size. After a discussion with others it was decided that it was not worth the effort right now.

**Comment by metricjs** [comment]
Can't see any other issues with the code so I would tick this except that the Jira number is incorrect

**Comment by HAFRMO** [comment]
Closing pull request to rename everything to the right number, will open a new one with everything correct. Sorry for the hassle.

**Pull request closed without merging**